src/Models/StepRegistry.cs

```diff
@@ -42,6 +44,44 @@ public void AddStep(string stepValue, GaugeMethod method)
             _registry.GetValueOrDefault(stepValue).Add(method);
         }
 
+        public void RemoveSteps(string filepath)
+        {
+            var newRegistry = new Dictionary<string, List<GaugeMethod>>();
+            foreach (var (key, gaugeMethods) in _registry)
+            {
+                var methods = gaugeMethods.Where(method => !filepath.Equals(method.FileName)).ToList();
+                if (methods.Count > 0) newRegistry[key] = methods;
+            }
+
+            _registry = newRegistry;
+        }
+
+        public IEnumerable<StepPosition> GetStepPositions(string filePath)
+        {
+            var positions = new List<StepPosition>();
+            foreach (var (stepValue, gaugeMethods) in _registry)
+                gaugeMethods.ForEach(m =>
```

Comment:
use same `foreach` syntax, like the previous line. In this case LINQ hurts readability IMO.